test-fork/UniswapMigrator.spec.ts

```diff
@@ -0,0 +1,296 @@
+        sushiOhmDaiLPToken = ERC20__factory.connect("0x055475920a8c93cffb64d039a8205f7acc7722d3", signer);
+    }
+
+    beforeEach(async () => {
+        await setup();
+    });
+
+    const expectMigrateUniswapV2AndJoinPool = async (
+        account: Account,
+        source: number,
+        tokens: Array<string>,
+        fromLPToken: ERC20,
+        toLPToken: ERC20,
+        rewardPool: ERC20,
+    ) => {
+        const isRewardPool = rewardPool.address !== ZERO_ADDRESS;
+
+        const fromLPPositionBefore = await fromLPToken.balanceOf(account.address);
+        const toLPPositionBefore = await toLPToken.balanceOf(account.address);
+        const rewardPoolBalanceBefore = isRewardPool ? await rewardPool.balanceOf(account.address) : BN.from(0);
+
+        expect(fromLPPositionBefore, "from position").to.gt(0);
+        const amountsMin = [1, 1];
+        const deadline = (await getTimestamp()).add(ONE_HOUR);
+        const [tokensSorted, amountsMinSorted] = balHelper.sortTokens(tokens, amountsMin);
+        const amountMinOut = 1;
+
+        await fromLPToken.connect(account.signer).approve(uniswapMigrator.address, ethers.constants.MaxUint256);
+
+        await uniswapMigrator.connect(account.signer).migrateUniswapV2AndJoinPool({
+            source,
+            fromLpToken: fromLPToken.address,
+            liquidity: fromLPPositionBefore,
+            tokens: tokensSorted,
+            amountsMin: amountsMinSorted as BN[],
+            deadline: deadline,
+            pool: toLPToken.address,
+            rewardPool: rewardPool.address,
+            amountMinOut: amountMinOut,
+        });
+
+        // after joining to the pool the balance is staked on aura therefore the following expectations
+        const fromLPPositionAfter = await fromLPToken.balanceOf(account.address);
+        const toLPPositionAfter = await toLPToken.balanceOf(account.address);
+
+        expect(fromLPPositionAfter, "from position").to.eq(0);
+
+        if (isRewardPool) {
+            const rewardPoolBalanceAfter = await rewardPool.balanceOf(account.address);
+            expect(rewardPoolBalanceAfter, "reward pool balance increases").to.gt(rewardPoolBalanceBefore);
+            expect(toLPPositionAfter, "balancer position does not change").to.eq(toLPPositionBefore);
+        } else {
+            expect(toLPPositionAfter, "balancer position increases change").to.gt(toLPPositionBefore);
+        }
+    };
+    const expectMigrateUniswapV2AndCreatePool = async (
+        account: Account,
+        source: number,
+        fromLPToken: ERC20,
+        poolData: PoolData,
+    ) => {
+        const fromLPPositionBefore = await fromLPToken.balanceOf(account.address);
+
+        expect(fromLPPositionBefore, "from position").to.gt(0);
+
+        await fromLPToken.connect(account.signer).approve(uniswapMigrator.address, ethers.constants.MaxUint256);
+        const amountsMin = [1, 1];
```

Comment:
We are not applying amountsMin correctly here. Should really pass values that we expect incase this leads to a false positive